tests/models/bark/test_modeling_bark.py

```diff
@@ -920,6 +920,27 @@ def test_generate_semantic(self):
 
         self.assertListEqual(output_ids[0, : len(expected_output_ids)].tolist(), expected_output_ids)
 
+    @slow
+    def test_generate_semantic_early_stop(self):
+        input_ids = self.inputs
+
+        # fmt: off
+        # check first ids
+        expected_output_ids = [7363, 321, 41, 1461, 6915, 952, 326, 41, 41, 927,]
+        # fmt: on
+
+        self.semantic_generation_config.min_eos_p = 0.05
+
+        with torch.no_grad():
+            output_ids = self.model.semantic.generate(
+                **input_ids,
+                do_sample=True,
+                temperature=1.0,
+                semantic_generation_config=self.semantic_generation_config,
+            )
+
+        self.assertNotEqual(output_ids[0, : len(expected_output_ids)].tolist()[-1], [-1])
```

Comment:
`output_ids` should always have `len(expected_output_ids)` right ?

Reply:
I was wondering about that. If early stopping works, wouldn't the output be shorter? The audio snippets generated with and without `min_eos_p` in the original repo's [example notebook](https://github.com/suno-ai/bark/blob/773624d26db84278a55aacae9a16d7b25fbccab8/notebooks/long_form_generation.ipynb#L160) had around 6 second difference over 70 seconds on the same input.

Reply:
The whole point of this is precisely to show that it has a different behaviour than without early stopping. So let's make this:
In the same test, generate with and without early stopping (with the same seed and a temperature!=1) and make sure that:
1. length of early stopping if strictly lower than without (it probably depends on the example and hyper-parameters you chose)
2. generated ids are the same on the whole generated ids of the early stopping one

Does that sound good?

Out of curiosity could you also put here the two audio snippets with and without early stopping with your implementaiton?

Reply:
Yep that sounds good (apologies, I need more coffee this morning). 

Here are the snippets from `test_generate_end_to_end_with_sub_models_args`:
[with_and_without.zip](https://github.com/huggingface/transformers/files/13041306/with_and_without.zip)
Notice that the one with min_eos_p is shorter (to me the difference in inflection makes it sound a little better).

Reply:
Really nice, thanks for the samples! did you use the same seed?

Reply:
Yes, set it to 0.  I'll add it back here. Took it out by mistake.